File: rust/src/value/x.rs
/// Background Colors.
pub struct X;

impl X {
    /// Black Color.
    pub const BLACK: [u8; 1]   = [40];
    /// Red Color.
    pub const RED: [u8; 1]     = [41];
    /// Green Color.
    pub const GREEN: [u8; 1]   = [42];
    /// Yellow Color.
    pub const YELLOW: [u8; 1]  = [43];
    /// Blue Color.
    pub const BLUE: [u8; 1]    = [44];
    /// Magenta Color.
    pub const MAGENTA: [u8; 1] = [45];
    /// Cyan Color.
    pub const CYAN: [u8; 1]    = [46];
    /// White Color.
    pub const WHITE: [u8; 1]   = [47];
    /// Default Color.
    pub const DEFAULT: [u8; 1] = [49];

    /// RGB Color.
    pub fn rgb(r: u8, g: u8, b: u8) -> [u8; 5] {
        [48, 2, r, g, b]
    }
}

File: javascript/src/value/x.ts
export class X {
    /** Black Color. */
    static BLACK: string   = "\x1b[40m"
    /** Red Color. */
    static RED: string     = "\x1b[41m"
    /** Green Color. */
    static GREEN: string   = "\x1b[42m"
    /** Yellow Color. */
    static YELLOW: string  = "\x1b[43m"
    /** Blue Color. */
    static BLUE: string    = "\x1b[44m"
    /** Magenta Color. */
    static MAGENTA: string = "\x1b[45m"
    /** Cyan Color. */
    static CYAN: string    = "\x1b[46m"
    /** White Color. */
    static WHITE: string   = "\x1b[47m"

    /** RGB Color. */
    static rgb(r: number, g: number, b: number): string {
        return `\x1b[48;2;${r};${g};${b}m`
    }
}